AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -579,7 +580,7 @@ - (void)openURLInAuthenticationSessionWith:(NSURL *)url API_AVAILABLE(ios(11)) {
     if (!strongSelf) {
       return;
     }
-    strongSelf.authenticationSession = nil;
+    [strongSelf clearAuthenticationSession];
```

Comment:
This method cancels the session but we don't need to cancel it at this point it's completed already right? Why not just breaking the strong reference as before?

Reply:
In the documentation of the 'cancel' method is written that if SafariViewController was presented, after the call 'cancel' it will be dismiss. At the time of completion handler execution this ViewController may still exist and we forcibly dismiss it before remove the reference.

Reply:
@clpolet No, it crashes in this case. it was fixed before by cleaning the reference in background event (that is not fully correct).

Reply:
Actually, we don't need to do any session clean up in the completion handler if we make sure to cancel the old session and reassign to the new session before starting the new session.